app/api/odm/ModelWithPermissions.ts

```diff
@@ -115,14 +120,15 @@ const controlPermissionsData = <T>(data: T & { published?: boolean }, user?: Use
 };
 
 export class ModelWithPermissions<T> extends OdmModel<T> {
-  async save(data: DataType<T & { permissions?: PermissionSchema[] }>) {
```

Comment:
I believe this was wrong to start with, the `& permissions` should have been OUTSIDE of the <T> bracket.

Reply:
im not sure this is the case with the current implementation, lets take a look together.